### PR TITLE
Allow find() with _id of ObjectId, UUID (in wrapped notation)

### LINKED ISSUE #924: Support find() with ObjectId / UUID - valued document ids (_id)
After #922 allows use of typed ObjectIds, UUIDS, as document ids (_id), we need to allow _id - based find() filtering using typed ObjectIds, UUIDs.

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**

Adds support for passing "JSON Extension values" as `_id` filter.

**Which issue(s) this PR fixes**:
Fixes #924

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/filter/FilterOperator.java [hunk 1]
```diff
@@ -37,12 +36,16 @@ private static void addComparisonOperator(FilterOperator filterOperator) {
...
-    public static FilterOperator getComparisonOperator(String operator) {
-      final FilterOperator filterOperator = operatorMap.get(operator);
-      if (filterOperator == null)
-        throw new JsonApiException(
-            ErrorCode.UNSUPPORTED_FILTER_OPERATION, "Unsupported filter operator " + operator);
+    public static FilterOperator findComparisonOperator(String operator) {
+      return operatorMap.get(operator);
+    }
 
+    @Deprecated // since 1.0.2 use "findComparisonOperator" instead
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/filter/FilterOperator.java:43]: Bad form to make caller catch exception, better to use `find()`. May delete before merging if things work as expected.

File: src/test/java/io/stargate/sgv2/jsonapi/api/model/command/deserializers/FilterClauseDeserializerTest.java [hunk 2]
```diff
@@ -1570,4 +1574,60 @@ public void invalidPathNameWithValidOperator() {
...
+            {"_id": {"$uuid": "%s"}}
+          """.formatted(UUID);
+      final ComparisonExpression expectedResult =
+          new ComparisonExpression(
+              "_id",
+              List.of(
+                  new ValueComparisonOperation(
+                      ValueComparisonOperator.EQ, new JsonLiteral(UUID, JsonType.DOCUMENT_ID))),
+              null);
+      FilterClause filterClause = objectMapper.readValue(json, FilterClause.class);
```
[the remark this answers is not in the document]
Comment (tatu-at-datastax) [reply src/test/java/io/stargate/sgv2/jsonapi/api/model/command/deserializers/FilterClauseDeserializerTest.java:1593]: yes, this is Draft, plenty more needs to be added.

If I ever get this working; logic in `FilterClauseDeserializer` is pretty muddled :-(
(spent lots of time trying to decipher logic, failing)

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/FindOneIntegrationTest.java [hunk 3]
```diff
@@ -769,15 +769,15 @@ public void findWithMissingCollection() {
           .body("data", is(nullValue()))
           .body("status", is(nullValue()))
           .body("errors", hasSize(1))
+          .body("errors[0].errorCode", is("COLLECTION_NOT_EXIST"))
```
Comment (tatu-at-datastax) [inline src/test/java/io/stargate/sgv2/jsonapi/api/v1/FindOneIntegrationTest.java:772]: Just reordering checks, no functional change

File: src/test/java/io/stargate/sgv2/jsonapi/api/model/command/deserializers/FilterClauseDeserializerTest.java [hunk 4]
```diff
@@ -1570,4 +1575,123 @@ public void invalidPathNameWithValidOperator() {
...
+                assertThat(t.getMessage())
+                    .contains(
+                        "Bad JSON Extension value: '$uuid' value has to be 36-character UUID String, instead got (\"abc\")");
+              });
+    }
+
+    @Test
+    public void mustFailOnBadObjectIdAsId() throws Exception {
+      String json = """
+         {"_id": {"$objectId": "xyz"}}
```
Comment (vkarpov15) [inline src/test/java/io/stargate/sgv2/jsonapi/api/model/command/deserializers/FilterClauseDeserializerTest.java:1651]: Minor detail: MongoDB extended JSON uses `$oid`, not `$objectId`. [Here's the internal JS implementation](https://github.com/mongodb/js-bson/blob/2ac17ec1e3c53b280efa298d137d96b2176bf046/src/extended_json.ts#L71) and [Extended JSON docs](https://www.mongodb.com/docs/manual/reference/mongodb-extended-json/#mongodb-bsontype-ObjectId).

`$oid` over `$objectId` isn't my favorite design decision, I can see the benefit of using `$objectId`. But I figured I'd surface this inconsistency so we can think a little more about whether we prefer `$oid` for consistency or `$objectId` for readability. Stargate-mongoose can do either way.

Comment (tatu-at-datastax) [reply src/test/java/io/stargate/sgv2/jsonapi/api/model/command/deserializers/FilterClauseDeserializerTest.java:1651]: Hmmh. I did miss this comment, for better or worse. So at least release 1.0.3 exposes `$objectId`; not sure we want to re-visit this.

/cc @amorton

**Pull request merged**